insights/core/remote_resource.py

```diff
@@ -0,0 +1,142 @@
+import requests
+
+import redis
+import calendar
+from cachecontrol.heuristics import BaseHeuristic
+from cachecontrol.wrapper import CacheControl
+from cachecontrol.caches.file_cache import FileCache
+
+from datetime import datetime, timedelta
+from email.utils import parsedate, formatdate
+from cachecontrol.caches.redis_cache import RedisCache
+
+
+class RemoteResource(object):
+    """
+    RemoteResource class for accessing external Web resources.
+
+    Attributes:
+        timeout(float): Time in seconds for the requests.get api call to wait before returning a timeout exception
+
+    Examples:
+        >>> from insights.core.remote_resource import RemoteResource
+        >>> rr = RemoteResource()
+        >>> rtn = rr.get("http://google.com")
+        >>> print (rtn.content)
+    """
+
+    timeout = 10
+
+    def __init__(self, session=None):
+
+        self.session = session or requests.Session()
+
+    def get(self, url, params={}, headers={}, auth=(), verify=False):
+        """
+        Returns the response payload from the request to the given URL.
+
+        Args:
+            url (str): The URL for the WEB API that the request is being made too.
+            params (dict): Dictionary containing the query string parameters.
+            headers (dict): HTTP Headers that may be needed forthe request.
+            auth (tuple): User ID and password for Basic Auth
+            verify (str/bool): Value must be path to the Cert Bundle if verifying the SSL certificate
+             or boolean False to ignore verifying the SSL certificate.
+
+        Returns:
+            Response:(Response): Response object from requests.get api request
+        """
+
+        return self.session.get(url, params=params, headers=headers, verify=verify, auth=auth,
+                            timeout=self.timeout)
+
+
+class CachedRemoteResource(RemoteResource):
+    """
+    RemoteResource subclass that sets up caching for subsequent Web resource requests.
+
+    Attributes:
+        expire_after (float): Amount of time in seconds that the cache will expire
+        backend (str): Type of storage for cache `DictCache1, `FileCache" or `RedisCache`
+        redis_host (str): Hpstname of redis instance if `RedisCache` backend is specified
+        redis_port (int): Port used to contact the redis instance if `RedisCache` backend is specified
+        file_cache_path (string): Path to where file cache will be stored if `FileCache` backend is specified
+        session (object): Requests session object
```

Comment:
Is this still an attribute?

Reply:
Removed. Made the change in the code review but forgot to remove from the doc